src/cdk/table/table.spec.ts

```diff
@@ -118,7 +118,7 @@ describe('CdkTable', () => {
         getRows(tableElement).forEach(row => {
           expect(row.getAttribute('role')).toBe('row');
           getCells(row).forEach(cell => {
-            expect(cell.getAttribute('role')).toBe('gridcell');
+            expect(cell.getAttribute('role')).toBe('cell');
```

Comment:
Can you add another unit test that asserts that the cell is `gridcell` when the table is changed to `grid`?

Reply:
I expanded a test case that set the role of the table to `treegrid`. Which got me thinking, should the cell roles be set to `gridcell` only if the table is set to `grid` or should it be set to `gridcell` if the table is anything other than `table` (eg. `treegrid`?)

Reply:
That's a good catch- we should set `gridcell` when the table is either `grid` or `treegrid` but leave it as `cell` in all other circumstances